x-pack/legacy/plugins/apm/public/components/app/Settings/CustomizeUI/CustomLink/CustomLinkFlyout/FiltersSection.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSelect,
+  EuiSpacer,
+  EuiText,
+  EuiTitle
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { isEmpty } from 'lodash';
+import React, { useRef } from 'react';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { FilterOptionsType } from '../../../../../../../../../../plugins/apm/server/routes/settings/custom_link';
+import {
+  getSelectOptions,
+  DEFAULT_OPTION,
+  filterSelectOptions,
+  Filters
+} from './helper';
+
+export const FiltersSection = ({
+  filters,
+  onChangeFilters
+}: {
+  filters: Filters;
+  onChangeFilters: (filters: Filters) => void;
+}) => {
+  const filterValueRefs = useRef<HTMLInputElement[]>([]);
+
+  const onChangeFilter = (filter: Filters[0], idx: number) => {
+    if (filterValueRefs.current[idx]) {
+      filterValueRefs.current[idx].focus();
+    }
+    const copyOfFilters = [...filters];
+    copyOfFilters[idx] = filter;
+    onChangeFilters(copyOfFilters);
+  };
+
+  const onRemoveFilter = (idx: number) => {
+    const copyOfFilters = [...filters];
+    copyOfFilters.splice(idx, 1);
+    // When empty, means that it was the last filter that got removed,
+    // so instead of showing an empty list, will add a new empty filter.
+    if (isEmpty(copyOfFilters)) {
+      copyOfFilters.push(['', '']);
+    }
+
+    onChangeFilters(copyOfFilters);
+  };
```

Comment:
Instead of this, what about disabling the "remove" icon on the last item?

Reply:
This is already the case when a user hasn't selected any filter. 
But the icon will be enabled as soon as something is selected, then in that case, if he clicks on the remove button, it will clean the fields.

Reply:
nvm, I see now see we want to reset it.

Minor changes with comments:
```ts
  const onRemoveFilter = (idx: number) => {
    // remove without mutating original array 
    const newFilters = [...filters].splice(idx, 1);

    // if there is only one item left it should not be removed 
    // but reset to empty
    if (isEmpty(newFilters)) {
      onChangeFilters(['', '']);
    } else {
      onChangeFilters(newFilters);
    }
  };
```

WDYT?